lib/src/utils/pnns_groups.dart

```diff
@@ -1,233 +1,276 @@
-  };
-
-  static const Map<PnnsGroup2, String> _IDS = {
-    PnnsGroup2.FRUIT_JUICES: 'fruit-juices',
-    PnnsGroup2.SWEETENED_BEVERAGES: 'sweetened-beverages',
-    PnnsGroup2.ALCOHOLIC_BEVERAGES: 'alcoholic-beverages',
-    PnnsGroup2.UNSWEETENED_BEVERAGES: 'unsweetened-beverages',
-    PnnsGroup2.ARTIFICIALLY_SWEETENED_BEVERAGES:
-        'artificially-sweetened-beverages',
-    PnnsGroup2.WATERS_AND_FLAVORED_WATERS: 'waters-and-flavored-waters',
-    PnnsGroup2.FRUIT_NECTARS: 'fruit-nectars',
-    PnnsGroup2.TEAS_AND_HERBAL_TEAS_AND_COFFEES:
-        'teas-and-herbal-teas-and-coffees',
-    PnnsGroup2.BREAKFAST_CEREALS: 'breakfast-cereals',
-    PnnsGroup2.PASTRIES: 'pastries',
-    PnnsGroup2.POTATOES: 'potatoes',
-    PnnsGroup2.CEREALS: 'cereals',
-    PnnsGroup2.BREAD: 'bread',
-    PnnsGroup2.PIZZA_PIES_AND_QUICHE: 'pizza-pies-and-quiche',
-    PnnsGroup2.SANDWICHES: 'sandwiches',
-    PnnsGroup2.SOUPS: 'soups',
-    PnnsGroup2.ONE_DISH_MEALS: 'one-dish-meals',
-    PnnsGroup2.DRESSINGS_AND_SAUCES: 'dressings-and-sauces',
-    PnnsGroup2.FATS: 'fats',
-    PnnsGroup2.PROCESSED_MEAT: 'processed-meat',
-    PnnsGroup2.MEAT: 'meat',
-    PnnsGroup2.FISH_AND_SEAFOOD: 'fish-and-seafood',
-    PnnsGroup2.OFFALS: 'offals',
-    PnnsGroup2.FRUITS: 'fruits',
-    PnnsGroup2.VEGETABLES: 'vegetables',
-    PnnsGroup2.DRIED_FRUITS: 'dried-fruits',
-    PnnsGroup2.NUTS: 'nuts',
-    PnnsGroup2.LEGUMES: 'legumes',
-    PnnsGroup2.MILK_AND_YOGURT: 'milk-and-yogurt',
-    PnnsGroup2.DAIRY_DESSERTS: 'dairy-desserts',
-    PnnsGroup2.CHEESE: 'cheese',
-    PnnsGroup2.PLANT_BASED_MILK_SUBSTITUTES: 'plant-based-milk-substitutes',
-    PnnsGroup2.APPETIZERS: 'appetizers',
-    PnnsGroup2.SALTY_AND_FATTY_PRODUCTS: 'salty-and-fatty-products',
-    PnnsGroup2.SWEETS: 'sweets',
-    PnnsGroup2.CHOCOLATE_PRODUCTS: 'chocolate-products',
-    PnnsGroup2.ICE_CREAM: 'ice-cream',
-  };
-
-  /// Returns the name of the subgroup, in English
-  String get name => _NAMES[this] ?? 'PNNS group 2';
+/// PNNS food subgroups.
+enum PnnsGroup2 implements OffTagged {
+  FRUIT_JUICES(
+    offTag: 'fruit-juices',
+    name: 'Fruit juices',
+  ),
+  SWEETENED_BEVERAGES(
+    offTag: 'sweetened-beverages',
+    name: 'Sweetened beverages',
+  ),
+  ALCOHOLIC_BEVERAGES(
+    offTag: 'alcoholic-beverages',
+    name: 'Alcoholic beverages',
+  ),
+  UNSWEETENED_BEVERAGES(
+    offTag: 'unsweetened-beverages',
+    name: 'Unsweetened beverages',
+  ),
+  ARTIFICIALLY_SWEETENED_BEVERAGES(
+    offTag: 'artificially-sweetened-beverages',
+    name: 'Artificially sweetened beverages',
+  ),
+  WATERS_AND_FLAVORED_WATERS(
+    offTag: 'waters-and-flavored-waters',
+    name: 'Waters and flavored waters',
+  ),
+  FRUIT_NECTARS(
+    offTag: 'fruit-nectars',
+    name: 'Fruits nectars',
+  ),
+  TEAS_AND_HERBAL_TEAS_AND_COFFEES(
+    offTag: 'teas-and-herbal-teas-and-coffees',
+    name: 'Teas and herbal teas and coffees',
+  ),
+
+  BREAKFAST_CEREALS(
```

Comment:
Why are there sometimes new lines between the subgroups and sometimes not

Reply:
> Why are there sometimes new lines between the subgroups and sometimes not

The usual question: the empty lines were there the way it was written previously (in that specific case, between level-1 groups of level-2 groups).
It doesn't seem to trigger lints warning so why not. Of course we could get rid of those useless empty lines.

Thank you @M123-dev for your review!